File: server/testdata/plugins/node-hello-plugin/fetch.mjs
import process from "node:process";

async function readJSON() {
  const chunks = [];
  for await (const chunk of process.stdin) {
    chunks.push(chunk);
  }

  const raw = Buffer.concat(chunks).toString("utf8").trim();
  return raw ? JSON.parse(raw) : {};
}

const payload = await readJSON();
const sourceName = String(payload.workspaceConfig?.sourceName ?? "Node Hello Source").trim();
const message = String(
  payload.scheduleConfig?.message ?? "Hello from the Node fixture plugin.",
).trim();
const tone = String(payload.scheduleConfig?.tone ?? "plain").trim();
const repeatValue = Number.parseInt(String(payload.scheduleConfig?.repeat ?? "1"), 10);
const repeat = Number.isFinite(repeatValue) && repeatValue > 0 ? repeatValue : 1;
const includeTriggeredAt = Boolean(payload.workspaceConfig?.includeTriggeredAt);

const repeated = Array.from({ length: repeat }, () => message);
if (tone === "verbose" && includeTriggeredAt && payload.trigger?.triggeredAt) {
  repeated.push(`Triggered at: ${payload.trigger.triggeredAt}`);
}

process.stdout.write(
  JSON.stringify({
    title: `${sourceName} Digest`,
    content: repeated.join("\n"),
    sourceLabel: sourceName,
  }),
);

File: server/testdata/plugins/python-hello-plugin/fetch.py
import json
import sys


def main() -> None:
    payload = json.loads(sys.stdin.read() or "{}")
    workspace_config = payload.get("workspaceConfig", {})
    trigger = payload.get("trigger", {}) or {}

    source_name = str(workspace_config.get("sourceName", "Python Hello Source")).strip()
    message = str(workspace_config.get("message", "Hello from the Python fixture plugin.")).strip()
    uppercase = bool(workspace_config.get("uppercase", False))

    body = message.upper() if uppercase else message
    external_id = f"python-hello-{trigger.get('triggeredAt', 'default')}"
    item = {
        "externalId": external_id,
        "title": f"{source_name} Digest",
        "sourceLabel": source_name,
        "blocks": [
            {"type": "heading", "level": 1, "text": f"{source_name} Digest"},
            {"type": "paragraph", "text": body},
        ],
    }

    sys.stdout.write(
        json.dumps(
            {
                "items": [item],
                "cursor": trigger.get("triggeredAt"),
            }
        )
    )


if __name__ == "__main__":
    main()
